lib/spack/spack/test/environment_modifications.py

```diff
@@ -406,6 +417,8 @@ def test_sanitize_regex(env, blacklist, whitelist, expected, deleted):
     assert all(x not in after for x in deleted)
 
 
+@pytest.mark.skipif(sys.platform == 'win32',
+                    reason="Not supported on Windows (yet)")
```

Comment:
A number of the failures in this module are indeed related to path inconsistencies, however, a further number still are caused because the modifications seemingly do not take, which is bizzare as the functionality works perfectly fine in production. I am looking into this, but may require some more in depth debugging.

Reply:
These tests are now passing w.r.t functionality, will refactor paths so tests can be re-enabled and pass. Pesky lack of os.pathsep useage.